src/main/kotlin/app/RepoHasher.kt

```diff
@@ -0,0 +1,89 @@
+// Copyright 2017 Sourcerer Inc. All Rights Reserved.
+// Author: Anatoly Kislov (anatoly@sourcerer.io)
+
+package app
+
+import app.model.Commit
+import app.model.Repo
+import app.utils.RepoHelper
+import io.reactivex.Observable
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.revwalk.RevWalk
+import java.io.File
+import java.io.IOException
+
+/**
+ * RepoHasher hashes repository and uploads stats to server.
+ */
+class RepoHasher(val repo: Repo) {
+    fun update() {
+        //TODO(anatoly): Implement repository analysis.
+        //TODO(anatoly): Implement data transfer.
+
+        var firstCommit: RevCommit
+
+        // Get commits loading observable and wait for subscribers.
+        val observableRevCommits = getObservableRevCommits().publish()
+
+        // Main data flow. Read commits from repo and send them to server.
+        observableRevCommits
+                .doOnNext {
```

Comment:
In many langs first method call in chain places at the same line as receiver, so i would write `observableRevCommits.doOnNext {`